arch/arc/core/switch.S

```diff
@@ -186,31 +183,8 @@ _switch_return_from_firq:
 #else
 	sr r3, [_ARC_V2_AUX_IRQ_ACT]
 #endif
-
-_switch_already_in_irq:
 	rtie
 
-.balign 4
-_return_from_exc_irq:
```

Comment:
Removing these 'return from exception' cases simplifies this code, good. From the commit description, I understand these cases should not happen as they cannot be handled (context wasn't saved). Can/should there be a check to ensure  these returns from exception (after context switch) do not happen, and raise a fatal error otherwise? Or, they cannot happen by design?

Reply:
good idea, i will consider it. In fact, then can not happen by desig, becasue switch.s belongs to cooperative scheduling. Before calling switch,  kernel will check whether it's interrupt context or exception mode. If yes, switch will not be called.

I thinks it's ok to add some extra codes to detect this case and raise an exception early

Reply:
ok - sounds like there's no need to spend cycles on an additional check.